File: Engine/Entities/Room.cs
namespace Engine.Entities
{
    // Chart of the Game
    // -----------------
    //
    //              3 (Blue2)
    // 4 (Red2)                       2 (Red1)
    //              1 (Blue1)          
    public class Room : BaseEntity
    {
        [Required]
        [StringLength(20)]
        public string GameType { get; set; }
        public int TargetScore { get; set; }
        public string HostName { get; set; }
        public string Blue1 { get; set; }
        public string Red1 { get; set; }
        public string Blue2 { get; set; }
        public string Red2 { get; set; }
        public bool Blue1Ready { get; set; }
        public bool Red1Ready { get; set; }
        public bool Blue2Ready { get; set; }
        public bool Red2Ready { get; set; }
        public DateTime Created { get; set; } = DateTime.UtcNow;
        public ICollection<Player> Players { get; set; } = new List<Player>();
    }
}

File: ClientApp/src/app/shared/models/engine/room.ts
import { Player } from "./player";

export class RoomToCreate {
    roomName: string;
    gameType: string;
    targetScore: number;

    constructor(roomName: string, gameType: string, targetScore: string) {
        this.roomName = roomName;
        this.gameType = gameType;
        this.targetScore = Number(targetScore);
    }
}

export interface RoomToJoin {
    roomName: string;
    gameType: string;
    hostName: string;
    targetScore: number;
    players: string[];
}

export interface Room {
    roomName: string;
    gameType: string;
    hostName: string;
    targetScore: number;
    blue1: string;
    red1: string;
    blue2: string;
    red2: string;
    blue1Ready: boolean;
    red1Ready: boolean;
    blue2Ready: boolean;
    red2Ready: boolean;
    readyButtonEnabled: boolean;
    players: Player[];
}